src/com/phonebookservice/util/MyArrayList.java

```diff
@@ -0,0 +1,416 @@
+
+        if (fromIndex == toIndex) {
+            return new MyArrayList<T>();
+        }
+
+        final List<T> subList = new MyArrayList<T>();
+
+        for (int i = 0; i < this.index; i++) {
+            if (i >= fromIndex && i < toIndex) {
+                subList.add(this.data[i]);
+            }
+
+            if (i > toIndex) {
+                break;
+            }
+        }
+        return subList;
+    }
+
+    /**
+     * Returns an array containing all of the elements in this list in proper
+     * sequence (from first to last element).
+     */
+    @Override
+    public T[] toArray() {
+        return this.copyArray(this.data, this.index);
+    }
+
+    /**
+     * Returns an array containing all of the elements in this list in proper
+     * sequence (from first to last element); the runtime type of the returned
+     * array is that of the specified array.
+     */
+    @Override
+    public <T> T[] toArray(final T[] array) {
+
+        if (array.length <= this.index) {
+            return (T[]) this.copyArray(this.data, this.index);
+        }
+
+        for (int i = 0; i < this.index; i++) {
+            array[i] = (T) this.data[i];
+        }
+
+        array[this.index] = null;
+        return array;
+    }
+
+    /**
+     * @return an iterator over the elements in this list in proper sequence.
+     */
+    @Override
+    public Iterator<T> iterator() {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Returns a list iterator over the elements in this list (in proper
+     * sequence).
+     */
+    @Override
+    public ListIterator<T> listIterator() {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Returns a list iterator over the elements in this list (in proper
+     * sequence), starting at the specified position in the list.
+     */
+    @Override
+    public ListIterator<T> listIterator(final int index) {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Removes the first occurrence of the specified element from this list, if
+     * it is present (optional operation).
+     */
+    @Override
+    public boolean remove(final Object element) {
+
+        if (element == null) {
+            for (int i = 0; i < this.index; i++) {
+                if (this.data[i] == null) {
+                    this.shiftRemove(i);
+                    return true;
+                }
+            }
+        } else {
+            for (int i = 0; i < index; i++) {
+                if (element.equals(this.data[i])) {
+                    this.shiftRemove(i);
+                    return true;
+                }
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * Removes the element at the specified index in this list.
+     *
+     * @param index the index.
+     */
+    @Override
+    public T remove(final int index) {
+        this.checkIndex(index);
+        T object = null;
+
+        if (this.data[index] != null) {
```

Comment:
This is tottaly wrong check.

If [1, null, 2]
I can list.remove (1)
This method prevent removing null and it shouldn't